lib/ansible/module_utils/facts/system/distribution.py

```diff
@@ -297,6 +293,9 @@ def parse_distribution_file_SUSE(self, name, data, path, collected_facts):
                         suse_facts['distribution_release'] = release.group(1)
                         suse_facts['distribution_version'] = collected_facts['distribution_version'] + '.' + release.group(1)
 
+        if os.path.islink('/etc/products.d/baseproduct') and os.path.realpath('/etc/products.d/baseproduct').endswith('SLES_SAP.prod'):
```

Comment:
Do you think we should add a backward compatible change for this fix? Also, please add a changelog entry and integration tests for this change? Thanks.

Reply:
Please keep the mock data files and add `/etc/products.d/baseproduct` file content in `input` section.

Something like ...
```
   "input": {
        "/etc/SuSE-release": "SUSE Linux Enterprise Server 12 (x86_64)\nVERSION = 12\nPATCHLEVEL = 2\n# This file is deprecated and will be removed in a future service pack or release.\n# Please check /etc/os-release for details about this release.",
        "/etc/os-release": "NAME=\"SLES_SAP\"\nVERSION=\"12-SP2\"\nVERSION_ID=\"12.2\"\nPRETTY_NAME=\"SUSE Linux Enterprise Server for SAP Applications 12 SP2\"\nID=\"sles_sap\"\nANSI_COLOR=\"0;32\"\nCPE_NAME=\"cpe:/o:suse:sles_sap:12:sp2\""
        "/etc/products.d/baseproduct": ...
    },
```

Reply:
@Akasurde this wouldn't make sense, because we don't parse the `/etc/products.d/baseproduct`file content, we check the `/etc/products.d/baseproduct` symlink target name